### PR TITLE
LockWorkStation and InitiateShutdown

### PR DESCRIPTION
**Pull request by Green7**

### CODE REVIEW

File: system.go [hunk 1]
```diff
@@ -22,6 +22,19 @@ import (
 	"github.com/winlabs/gowin32/wrappers"
 )
 
+func InitiateShutdown(machineName string, message string, gracePriod int, shutdownFlags int, reason int) error {
```
Comment (dgolub) [inline system.go]: [Nitpick] `gracePriod` -> `gracePeriod`.

File: system.go [hunk 2]
```diff
@@ -22,6 +22,19 @@ import (
 	"github.com/winlabs/gowin32/wrappers"
 )
 
+func InitiateShutdown(machineName string, message string, gracePriod int, shutdownFlags int, reason int) error {
+	if err := wrappers.InitiateShutdown(
```
Comment (dgolub) [inline system.go]: Also, let's add enumerations for the `shutdownFlags` and `reason` fields.

Comment (Green7) [reply system.go]: what about recent changes?

File: system.go [hunk 3]
```diff
@@ -22,6 +22,81 @@ import (
...
+	InitiateShutdownReasonMinorHung                 InitiateShutdownReason = wrappers.SHTDN_REASON_MINOR_HUNG
+	InitiateShutdownReasonMinorUnstable             InitiateShutdownReason = wrappers.SHTDN_REASON_MINOR_UNSTABLE
+	InitiateShutdownReasonMinorDisk                 InitiateShutdownReason = wrappers.SHTDN_REASON_MINOR_DISK
+	InitiateShutdownReasonMinorProcessor            InitiateShutdownReason = wrappers.SHTDN_REASON_MINOR_PROCESSOR
+	InitiateShutdownReasonMinorNetwordCard          InitiateShutdownReason = wrappers.SHTDN_REASON_MINOR_NETWORKCARD
+	InitiateShutdownReasonMinorPowerSupply          InitiateShutdownReason = wrappers.SHTDN_REASON_MINOR_POWER_SUPPLY
+	InitiateShutdownReasonMinorCordunplugged        InitiateShutdownReason = wrappers.SHTDN_REASON_MINOR_CORDUNPLUGGED
+	InitiateShutdownReasonMinorEnvironment          InitiateShutdownReason = wrappers.SHTDN_REASON_MINOR_ENVIRONMENT
+	InitiateShutdownReasonMinorHardwareDriver       InitiateShutdownReason = wrappers.SHTDN_REASON_MINOR_HARDWARE_DRIVER
+	InitiateShutdownReasonMinorOtherdriver          InitiateShutdownReason = wrappers.SHTDN_REASON_MINOR_OTHERDRIVER
```
Comment (dgolub) [inline system.go]: [Minor] `Otherdriver` -> `OtherDriver`.

Similarly, `Bluescreen` -> `BlueScreen`, `Securityfix` -> `SecurityFix`, etc. etc.

Comment (Green7) [reply system.go]: ok. I did the capitalization in a similar way to the underscores in MS constansts. But you right it looks bad.

File: system.go [hunk 4]
```diff
@@ -22,6 +22,81 @@ import (
 	"github.com/winlabs/gowin32/wrappers"
 )
 
+type InitiateShutdownFlag uint32
```
Comment (dgolub) [inline system.go]: `InitiateShutdownFlag` -> `InitiateShutdownFlags` for consistency with existing data types.

File: system.go [hunk 5]
```diff
@@ -22,6 +22,81 @@ import (
...
+
+const (
+	InitiateShutdownReasonMajorOther           InitiateShutdownReason = wrappers.SHTDN_REASON_MAJOR_OTHER
+	InitiateShutdownReasonMajorHardware        InitiateShutdownReason = wrappers.SHTDN_REASON_MAJOR_HARDWARE
+	InitiateShutdownReasonMajorOperatingSystem InitiateShutdownReason = wrappers.SHTDN_REASON_MAJOR_OPERATINGSYSTEM
+	InitiateShutdownReasonMajorSoftware        InitiateShutdownReason = wrappers.SHTDN_REASON_MAJOR_SOFTWARE
+	InitiateShutdownReasonMajorApplication     InitiateShutdownReason = wrappers.SHTDN_REASON_MAJOR_APPLICATION
+	InitiateShutdownReasonMajorSystem          InitiateShutdownReason = wrappers.SHTDN_REASON_MAJOR_SYSTEM
+	InitiateShutdownReasonMajorPower           InitiateShutdownReason = wrappers.SHTDN_REASON_MAJOR_POWER
+	InitiateShutdownReasonMajorLegacyApi       InitiateShutdownReason = wrappers.SHTDN_REASON_MAJOR_LEGACY_API
```
Comment (dgolub) [inline system.go]: `LegacyApi` -> `LegacyAPI`

File: system.go [hunk 6]
```diff
@@ -22,6 +22,81 @@ import (
...
+	InitiateShutdownReasonMinorEnvironment          InitiateShutdownReason = wrappers.SHTDN_REASON_MINOR_ENVIRONMENT
+	InitiateShutdownReasonMinorHardwareDriver       InitiateShutdownReason = wrappers.SHTDN_REASON_MINOR_HARDWARE_DRIVER
+	InitiateShutdownReasonMinorOtherDriver          InitiateShutdownReason = wrappers.SHTDN_REASON_MINOR_OTHERDRIVER
+	InitiateShutdownReasonMinorBlueScreen           InitiateShutdownReason = wrappers.SHTDN_REASON_MINOR_BLUESCREEN
+	InitiateShutdownReasonMinorServicePack          InitiateShutdownReason = wrappers.SHTDN_REASON_MINOR_SERVICEPACK
+	InitiateShutdownReasonMinorHotFix               InitiateShutdownReason = wrappers.SHTDN_REASON_MINOR_HOTFIX
+	InitiateShutdownReasonMinorSecurityFix          InitiateShutdownReason = wrappers.SHTDN_REASON_MINOR_SECURITYFIX
+	InitiateShutdownReasonMinorSecurity             InitiateShutdownReason = wrappers.SHTDN_REASON_MINOR_SECURITY
+	InitiateShutdownReasonMinorConnectivity         InitiateShutdownReason = wrappers.SHTDN_REASON_MINOR_NETWORK_CONNECTIVITY
+	InitiateShutdownReasonMinorWmi                  InitiateShutdownReason = wrappers.SHTDN_REASON_MINOR_WMI
```
Comment (dgolub) [inline system.go]: `ReasonMinorWmi` -> `ReasonMinorWMI`

File: system.go [hunk 7]
```diff
@@ -22,6 +22,81 @@ import (
...
+	InitiateShutdownReasonMinorServicePackUninstall InitiateShutdownReason = wrappers.SHTDN_REASON_MINOR_SERVICEPACK_UNINSTALL
+	InitiateShutdownReasonMinorHotfixUninstall      InitiateShutdownReason = wrappers.SHTDN_REASON_MINOR_HOTFIX_UNINSTALL
+	InitiateShutdownReasonMinorSecurityFixUninstall InitiateShutdownReason = wrappers.SHTDN_REASON_MINOR_SECURITYFIX_UNINSTALL
+	InitiateShutdownReasonMinorMmc                  InitiateShutdownReason = wrappers.SHTDN_REASON_MINOR_MMC
```
Comment (dgolub) [inline system.go]: `ReasonMinorMmc` -> `ReasonMinorMMC`

File: system.go [hunk 8]
```diff
@@ -22,6 +22,81 @@ import (
...
+	InitiateShutdownFlagInstallUpdates     InitiateShutdownFlags = wrappers.SHUTDOWN_INSTALL_UPDATES
+	InitiateShutdownFlagRestartApps        InitiateShutdownFlags = wrappers.SHUTDOWN_RESTARTAPPS
+	InitiateShutdownFlagSkipSvcPreshutdown InitiateShutdownFlags = wrappers.SHUTDOWN_SKIP_SVC_PRESHUTDOWN
+	InitiateShutdownFlagHybrid             InitiateShutdownFlags = wrappers.SHUTDOWN_HYBRID
+)
+
+type InitiateShutdownReason uint32
+
+const (
+	InitiateShutdownReasonUserDefinied InitiateShutdownReason = wrappers.SHTDN_REASON_FLAG_USER_DEFINED
```
Comment (dgolub) [inline system.go]: Missed this the first time through: `UserDefinied` -> `UserDefined`

File: system.go [hunk 9]
```diff
@@ -22,6 +22,81 @@ import (
...
+	InitiateShutdownReasonPlaned       InitiateShutdownReason = wrappers.SHTDN_REASON_FLAG_PLANNED
```
Comment (dgolub) [inline system.go]: Missed this one, too: `Planed` -> `Planned`

**Pull request merged**